pkgs/development/python-modules/petsc4py/default.nix

```diff
@@ -0,0 +1,46 @@
+{ lib
+, fetchPypi
+, python
+, buildPythonPackage
+, cython
+, numpy
+, petsc
+, openmpi
+, openssh
+, mpi4py
+, pytest }:
+
+buildPythonPackage rec {
+  pname = "petsc4py";
+  version = "3.10.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "094hcnran0r2z1wlvmjswsz3ski1m9kqrl5l0ax8jjhnk55x0flh";
+  };
+
+  PETSC_DIR = "${petsc}";
+
+  setupPyBuildFlags = [ "build_src --force" ];
+
+  buildInputs = [
+    cython
+    numpy
+    openssh
+    mpi4py
+  ];
+
+  propagatedBuildInputs = [
+    petsc
+    openmpi
```

Comment:
I don't find `openmpi` in ther setup.py.
`petsc` can probably be dropped from here since you specify `PETSC_DIR`.

Reply:
I moved `openmpi` to `nativeBuildInputs` and removed `petsc`.  The package doesn't build without `openmpi`.